### PR TITLE
Fixes airlocks that have panel shielding being un-repairable with a welder

### LINKED ISSUE #13388: Can't repair reinforced airlocks
**Issue Description**:

Can't repair airlocks which has the wires reinforced, instead, it tries to cut through the panel shielding, though the maintenance panel is closed

**What did you expect to happen**: 

Airlock gets repaired.

**What happened instead**:

Uses up fuel and gives the message from cutting through the panels shielding, but the shielding does not get affected.

**Why is this bad/What are the consequences**:

Can't repair more secure doors

**Steps to reproduce the problem**:

Damage a secure door, like the AI upload, attempt to repair it

**When did the problem start happening**:

Likely since the reinforcement of the airlocks were merged.

### PR DESCRIPTION
**Pull request by SteelSlayer**
## What Does This PR Do
Panel shielding meaning that is has a metal/plasteel sheet added to the airlock that you need to slice off before you could hack it. Includes stuff like high security airlocks, or vault airlocks, etc.

Issue was we weren't checking if the airlock's panel was open before removing shielding, so you could attempt to remove it even while the airlock's panel was closed, which blocked any repair attempts.

## Why It's Good For The Game
Fixes #13388

## Changelog
:cl:
fix: Fixes airlocks that have panel shielding being un-repairable with a welder
/:cl:

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
@Citinited

File: code/game/machinery/doors/airlock.dm [hunk 1]
```diff
@@ -965,63 +965,58 @@ About the new airlock wires panel:
...
-			else
-				to_chat(user, "<span class='notice'>The airlock doesn't need repairing.</span>")
+		else if(obj_integrity < max_integrity)
+			user.visible_message("[user] is welding the airlock.", \
+				"<span class='notice'>You begin repairing the airlock...</span>", \
+				"<span class='italics'>You hear welding.</span>")
+			if(I.use_tool(src, user, 40, volume = I.tool_volume, extra_checks = CALLBACK(src, .proc/weld_checks, I, user)))
+				obj_integrity = max_integrity
+				stat &= ~BROKEN
+				user.visible_message("[user.name] has repaired [src].", \
```
Comment (farie82) [inline code/game/machinery/doors/airlock.dm:1015]: No span class here

File: code/game/machinery/doors/airlock.dm [hunk 2]
```diff
@@ -965,63 +965,58 @@ About the new airlock wires panel:
...
+				if(!density && !welded)
+					return
+				welded = !welded
+				user.visible_message("[user.name] has [welded? "welded shut":"unwelded"] [src].", \
+					"<span class='notice'>You [welded ? "weld the airlock shut":"unweld the airlock"].</span>")
 				update_icon()
-			else
-				to_chat(user, "<span class='notice'>The airlock doesn't need repairing.</span>")
+		else if(obj_integrity < max_integrity)
+			user.visible_message("[user] is welding the airlock.", \
```
Comment (farie82) [inline code/game/machinery/doors/airlock.dm:1009]: No span class here

File: code/game/machinery/doors/airlock.dm [hunk 3]
```diff
@@ -965,63 +965,58 @@ About the new airlock wires panel:
...
+		if(user.a_intent != INTENT_HELP)
+			user.visible_message("[user] is [welded ? "unwelding":"welding"] the airlock.", \
+				"<span class='notice'>You begin [welded ? "unwelding":"welding"] the airlock...</span>", \
+				"<span class='italics'>You hear welding.</span>")
+
+			if(I.use_tool(src, user, 40, volume = I.tool_volume, extra_checks = CALLBACK(src, .proc/weld_checks, I, user)))
+				if(!density && !welded)
+					return
+				welded = !welded
+				user.visible_message("[user.name] has [welded? "welded shut":"unwelded"] [src].", \
```
Comment (farie82) [inline code/game/machinery/doors/airlock.dm:1005]: No span class here

File: code/game/machinery/doors/airlock.dm [hunk 4]
```diff
@@ -965,63 +965,58 @@ About the new airlock wires panel:
...
+				to_chat(user, "<span class='notice'>You begin cutting the inner layer of shielding...</span>")
+				if(!I.use_tool(src, user, 40, volume = I.tool_volume))
+					return
+				user.visible_message("<span class='notice'>[user] cuts through \the [src]'s shielding.</span>",
+					"<span class='notice'>You cut through \the [src]'s shielding.</span>",
+					"<span class='italics'>You hear welding.</span>")
+				security_level = AIRLOCK_SECURITY_PLASTEEL_I_S
+	else
+		if(user.a_intent != INTENT_HELP)
+			user.visible_message("[user] is [welded ? "unwelding":"welding"] the airlock.", \
```
Comment (farie82) [inline code/game/machinery/doors/airlock.dm:997]: No span class here

File: code/game/machinery/doors/airlock.dm (hunk 1 above)
Comment (SteelSlayer) [reply code/game/machinery/doors/airlock.dm:1015]: Sure, why not, I'll get all these spans

**Pull request merged**